src/Val.cc

```diff
@@ -3248,7 +3239,11 @@ PortVal* ValManager::GetPort(uint32_t port_num, TransportProto port_type) const
 		port_num = 0;
 		}
 
-	auto rval = ports[port_type][port_num];
+	auto &slot = const_cast<PortVal *&>(ports[port_type][port_num]); // TODO: fix const_cast
```

Comment:
TODO?

Also, as a style nit, type modifiers like `&` and `*` should be directly attached to the type and not the variable..

Reply:
Oh yes, I forgot to come up with something without `const_cast`, an operator which I very much try to avoid.
I'll fix the style. I'm used to attaching those modifiers to the variable name, because of C's syntax weirdness:
```
int* a, b;
```
Now what is `b`'s type? The type looks like `int*`, but the `*` only applies to `a`. But like this, it's somewhat more obvious:
```
int *a, b;
```
With your coding style, how do express this?
```
int a, *b;
```

Reply:
I'd suggest to just avoid the potential confusion by preferring to declare names separately (or at least require the separation for pointers/references).  Think either people already gravitate toward doing that or it's a rare situation anyway.